plugins/mapper-annotated-text/src/main/java/org/elasticsearch/index/mapper/annotatedtext/AnnotatedTextFieldMapper.java

```diff
@@ -570,39 +568,23 @@ public FieldMapper.Builder getMergeBuilder() {
     }
 
     @Override
-    protected SyntheticSourceMode syntheticSourceMode() {
-        return SyntheticSourceMode.NATIVE;
-    }
-
-    @Override
-    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
-        if (copyTo().copyToFields().isEmpty() != true) {
-            throw new IllegalArgumentException(
-                "field [" + fullPath() + "] of type [" + typeName() + "] doesn't support synthetic source because it declares copy_to"
-            );
-        }
+    protected SyntheticSourceSupport syntheticSourceSupport() {
         if (fieldType.stored()) {
-            return new StringStoredFieldFieldLoader(fullPath(), leafName()) {
+            var loader = new StringStoredFieldFieldLoader(fullPath(), leafName()) {
                 @Override
                 protected void write(XContentBuilder b, Object value) throws IOException {
                     b.value((String) value);
                 }
             };
+
+            return new SyntheticSourceSupport.Native(loader);
         }
 
         var kwd = TextFieldMapper.SyntheticSourceHelper.getKeywordFieldMapperForSyntheticSource(this);
         if (kwd != null) {
-            return kwd.syntheticFieldLoader(leafName());
+            return new SyntheticSourceSupport.Native(kwd.syntheticFieldLoader(leafName()));
```

Comment:
Maybe we can consider naming the two synthetic source modes `WithDocValues` and `WithStoredFields` instead of `Native` and non native? Like:
* `SyntheticSourceSupport.WithDocValues`
* `SyntheticSourceSupport.WithStoredFields`

Reply:
That's not really the distinction i am trying to make. Yes, `_ignored_source` is a stored field that holds the value used for synthetic source but it is also possible that the field itself uses a _different_ stored field for synthetic source. That becomes pretty blurry.

The difference between `Native` and `Fallback` is who controls synthetic source logic. In `Native` it's the mapper, in `Fallback` it's on a layer above and the mapper does not control what happens.